### PR TITLE
chore: refactor dropdown mixins (#1463)

### LINKED ISSUE #1463: Listbox & Menu: standalone versions should not have border radius
Due to the way the mixins are organised, standalone listboxes and menus are inheriting the border radius that is intended only for the button flyout versions. 

Solution would be to rework the current dropdown mixins into two: dropdown-mixins (for anything in a button flyout) and selection-list-mixins (for anything non-flyout).

The dropdown-mixins would most likely be able to inherit much of the base styles defined in selection-list-mixins.

Screen Shot 2021-06-04 at 10 16 08 AM

### PR DESCRIPTION
**Pull request by ianmcburnie**
Closes #1463

Listbox and menu modules were using the same "dropdown" mixins as listbox-button and menu-button. The problem is that listbox and menu are not considered "dropdowns", and can be deployed directly onto the page without any button (most often this happens inside of panel dialogs) and so we are going to run into issues with them sharing the exact same mixins and vars.

I have reorganised like so:

* dropdown: now refers only to the dropdown overlay/flyout
* selection-list-item: new generic term for an `option` or `menuitem`
* option: new specific term for items in any aria listbox 
* menuitem: new specific term for items in any aria menu

In a nutshell, most of the dropdown vars and mixins have been moved over to selection-list vars and mixins. Most option and menuitem mixins/vars are simple aliases to selection-list-item.

Not expecting reviewer to look through every file or line here, just a quick scan of this description and see if direction makes sense.

### CODE REVIEW

**Review by austingardner — APPROVED** [review]
LGTM! Having the same mixins for all of the dropdown items makes sense

File: src/less/variables/base/menu-variables.less [hunk 1]
```diff
@@ -1 +1,15 @@
-@menu-item-disabled-foreground-color: @color-text-disabled;
+@menu-background-color: @selection-list-background-color;
```
Comment (agliga) [inline src/less/variables/base/menu-variables.less:1]: Is there a reason why we need to define all these variables again and just point them to `selection-list`? Why not use `selection-list` directly?
Seems a bit overkill to redefine all these variables. I can see it for organization, but I think it gets confusing what the original color when you have like 3 levels of variable definitions (base color > selection-list > menuitem/listbox-option)

I think the only time we should define a variable is when one of these components needs it's own color different than the others. But for now if we add a new color or something for all dropdowns we have to maintain it in like 3+ different places
I think calling it `item` directly makes it just as intuitive and generic for both listbox and menu.

**Review by agliga — COMMENTED** [review]
Looks good overall. One comment about variable naming.

File: src/less/variables/base/menu-variables.less (hunk 1 above)
Comment (ianmcburnie) [reply src/less/variables/base/menu-variables.less:1]: Yeah, that's a fair comment and good observation. I did have it that way originally - just pointing to the `selection-list-item` variables directly as you say. However, I felt it might not be too obvious what a 'selection-list' is when looking through the code, and so terms like `menu-menuitem` and `listbox-option` seem to hammer home the difference.

If we add a new color, we shouldn't have to update it in 3 places though, so I'm not sure what you mean. We would only need to change the colour at the `selection-list` level, right? If you are correct, then I will consider reverting, because that wouldn't be good.

**Review by agliga — APPROVED** [review]
